bynder_sdk/client/upload_client.py

```diff
@@ -1,144 +1,129 @@
-class UploadClient():
+class UploadClient:
     """ Client to upload asset to Bynder.
     """
+    file_sha256 = ''
+
     def __init__(self, session):
         self.session = session
 
     def upload(self, file_path, media_id, upload_data):
-        """ Handles the upload of the file.
+        """Handles the upload of the file.
+        :param media_id: The media_id of the asset to be created or updated.
+        :param file_path: The path of the asset to be uploaded.
+        :param upload_data: The upload_data containing brand_id that can be
+        used in the save media endpoint.
+        :return: A dict with the keys:
+                - file_id: an identifier for the file uploaded
+                - correlation_id: The correlation_id of save media request.
+                - media: a dict containing save media related info.
         """
-        init_data, total_parts = self._run_s3_upload(file_path)
-        finalise_data = self._finalise_file(init_data, total_parts)
-        return self._save_media(finalise_data['importId'],
-                                upload_data, media_id)
-
-    def _run_s3_upload(self, file_path):
-        """ Uploads the media to Amazon S3 bucket-endpoint.
-        """
-        with open(file_path, 'rb') as f:
-            part_nr = 0
-            total_parts = math.ceil(
-                os.stat(f.fileno()).st_size / MAX_CHUNK_SIZE)
-
-            filename = file_path.rsplit('/', 1)[-1]
-            build_part_data = self._init_upload(filename, total_parts)
-
-            part_bytes = f.read(MAX_CHUNK_SIZE)
-            while part_bytes:
-                part_nr = part_nr + 1
-                part_data = build_part_data(part_nr)
-
-                response = self.session.post(
-                    self.upload_url,
-                    files={'file': part_bytes},
-                    data=part_data['multipart_params'],
-                )
-                response.raise_for_status()
-                self._register_part(part_data, part_nr)
-                part_bytes = f.read(MAX_CHUNK_SIZE)
-
-        return part_data, total_parts
-
-    def _init_upload(self, filename, total_parts):
-        """ Gets the URL of the Amazon S3 bucket-endpoint in the region
-        closest to the server and initialises a file upload with Bynder
-        and returns authorisation information to allow uploading to the
-        Amazon S3 bucket-endpoint.
+        try:
+            _, file_name = os.path.split(file_path)
+            file_id = self._prepare()
+            with open(file_path, "rb") as f:
+                self.file_sha256 = sha256(f.read()).hexdigest()
+            chunks_count, file_size = self._upload_chunks(file_path, file_id)
+            correlation_id = self._finalise_file(file_id, file_name, file_size,
+                                                 chunks_count)
+            media = self._save_media(file_id, upload_data, media_id)
+            return {'file_id': file_id, 'correlation_id': correlation_id,
```

Comment:
My understanding was the user can get back to us with a correlation Id and the file id in case something goes wrong with the upload to help diagnose the issue. But I can get rid of it for consistency with the other sdks.